recruit/candidates.go

```diff
@@ -0,0 +1,507 @@
+package recruit
+
+import (
+	"fmt"
+	"time"
+
+	zoho "github.com/schmorrison/Zoho"
+)
+
+// https://www.zoho.eu/recruit/developer-guide/apiv2/insert-records.html
+func (c *API) InsertCandidates(request InsertCandidateRequest) (data InsertCandidateResponse, err error) {
+	endpoint := zoho.Endpoint{
+		Name:         "InsertCandidates",
+		URL:          fmt.Sprintf("https://recruit.zoho.%s/recruit/v2/%s", c.ZohoTLD, CandidatesModule),
+		Method:       zoho.HTTPPost,
+		ResponseData: &InsertCandidateResponse{},
+		RequestBody:  request,
+	}
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return InsertCandidateResponse{}, fmt.Errorf("failed to insert Candidate(s): %s", err.Error())
+	}
+
+	if v, ok := endpoint.ResponseData.(*InsertCandidateResponse); ok {
+		for _, resp := range v.Data {
+			if resp.Code != "SUCCESS" {
+				return InsertCandidateResponse{}, fmt.Errorf("failed to insert Candidate(s): %s: %s", resp.Code, resp.Details)
+			}
+		}
+		return *v, nil
+	}
```

Comment:
This is fine for now, but in the future we """Might"""" need to handle this more carefully. I'm not sure how we handle such errors elsewhere. This type of fail could lead to inconsistencies in a user application.

Maybe we can add a field or method in the response struct like: ```Success bool `json:"-"```` so a user can easily double check for failed events without needing to  have knowledge of all endpoint particulars. This is probably a good item for another PR.